lib/health_cards/health_card_constraints.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'zlib'
+require 'json/minify'
+
+module HealthCards
+  ## Functionality for "Health Cards are Small" section of the Smart Health Cards Specification
+  module HealthCardConstraints
+    def bundle_operations(payload)
+      bundle = payload['vc']['credentialSubject']['fhirBundle']
+      if bundle
+        entries = bundle['entry']
+        entries, url_map = redefine_uris(entries)
+
+        update_elements(entries, url_map)
+      end
+      payload
+    end
+
+    def redefine_uris(resources)
+      url_map = {}
+      resource_count = 0
+
+      resources.each do |entry|
+        old_url = entry['fullUrl']
+        new_url = "resource:#{resource_count}"
+
+        url_map[old_url] = new_url
+        entry['fullUrl'] = new_url
+
+        resource_count += 1
+      end
+
+      [resources, url_map]
+    end
+
+    def update_elements(resources, url_map)
+      resources.each do |entry|
+        resource = entry['resource']
+
+        resource.delete('id')
+        resource.delete('meta')
+        resource.delete('text')
+        update_nested_elements(resource, url_map)
+      end
+      resources
+    end
+
+    def update_nested_elements(hash, url_map) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```

Comment:
Can this be refactored or split up into methods? I'm not sure disabling rubicon here is a good idea

Reply:
Added ticket [DVCI-53](https://jira.mitre.org/browse/DVCI-53) to update the way that FHIR Resources are navigated. According to our discussion today, I'll mark this as resolved and we can pick this back up after current PRs are merged.